react/src/components/ModelTryContentButton.tsx

```diff
@@ -236,33 +230,143 @@ const ModelTryContentButton: React.FC<ModelTryContentButtonProps> = ({
+      envvars: [],
       enabledAutomaticShmem: false,
     };
   };
 
-  const cloneOrCreateModelService = (runtimeVariant: string) => {
+  // Check if folder has required definition files
+  const checkFolderDefinitionFiles = async (vfolderId: string) => {
+    try {
+      const filesResponse = await baiClient.vfolder.list_files('', vfolderId);
+      const files = filesResponse?.items || [];
+      const hasModelDefinition = _.some(
+        files,
+        (file: { name?: string }) => file?.name === 'model-definition.yaml',
+      );
+      const hasServiceDefinition = _.some(
+        files,
+        (file: { name?: string }) => file?.name === 'service-definition.toml',
+      );
+
+      return {
+        hasModelDefinition,
+        hasServiceDefinition,
+        files,
+      };
+    } catch (error) {
+      console.error('Error checking folder files:', error);
+      // Provide more specific error information
+      const errorMessage =
+        error instanceof Error ? error.message : 'Unknown error occurred';
+      console.error('Detailed error:', errorMessage);
+      return {
+        hasModelDefinition: false,
+        hasServiceDefinition: false,
+        files: [],
+        error: errorMessage,
+      };
+    }
+  };
+
+  // Helper function to create service notification
+  const createServiceNotification = (
+    result: ServiceResult,
+    modelId: string,
+  ) => {
+    upsertNotification({
+      key: result?.endpoint_id,
+      open: true,
+      message: t('modelService.StartingModelService'),
+      duration: 0,
+      backgroundTask: {
+        promise: new Promise<void>((resolve, reject) => {
+          let progress = 0;
+          const interval = setInterval(async () => {
+            try {
+              progress += _.random(2, 5);
+              upsertNotification({
+                key: result?.endpoint_id,
+                backgroundTask: {
+                  status: 'pending',
+                  percent: progress > 100 ? 100 : progress,
+                },
+              });
+              const routingStatus = await baiRequestWithPromise({
+                method: 'GET',
+                url: `/services/${result?.endpoint_id}`,
+              });
+              if (routingStatus.active_routes.length > 0) {
+                clearInterval(interval);
+                return resolve();
+              }
+              if (progress >= 100) {
+                throw new Error(t('modelService.ModelServiceFailedToStart'));
+              }
+            } catch (error) {
+              clearInterval(interval);
+              return reject();
+            }
+          }, 5000);
+        }),
+        onChange: {
+          pending: t('modelService.StartingModelService'),
+          resolved: {
+            duration: 0,
+            open: true,
+            key: result?.endpoint_id,
+            backgroundTask: {
+              status: 'resolved',
+              percent: 100,
+            },
+            message: '',
```

Comment:
BAINotification uses merge to manage notifications with the same key. If you set message to an empty string, the merge process will assign an existing string and the message will not change.

message must be set to accept null.[ref](https://github.com/lablup/backend.ai-webui/pull/4050/files#diff-5ae227ac1cfa66a1771ec7a643f0ee04081657ecd9769700e6662bdc3a3811b8R55-R56)